### ISSUE TITLE
Extensions supported

### ISSUE THREAD

**Issue by matthewhanson** [radiantearth/stac-spec#278]
It would be useful to know the extensions that are supported by a given collection. In one catalog you may have a collection of EO data, and another collection of pointcloud data. 

I suggest an optional extensions field at the collection level.

```
extensions = ['eo', 'commons']
```

**Comment by m-mohr** [comment]
Could you elaborate more when this is useful and where to put this? Would this be added to the collection link in the item? Or would this be added to the catalog/collection json file itself? If the latter, you could just figure it out by checking for required fields, right? Other than that: What are the names that you are using? eo and commons are not directly specified somewhere. One could get the idea that these values are the prefixes, but commons doesn't has a prefix. How would you reference the datetime range extension (see #274)? dtr (prefix)? datetime-range (folder name), ...?

**Comment by matthewhanson** [comment]
@m-mohr Maybe something in the links would be a better way too, a rel="extension" with a link to the extension definition.

The names I used above would be an extension ID, but those aren't currently defined.

You could check the fields in the Items, but it's a little awkward. You would have to iterate through all the attributes, split them on ':', then get the set of them to know which extensions are used.

One use case is this: I've got a static catalog of collections, and want to ingest them into some other system but I only support certain extensions. 

I was thinking another reason would be if I wanted to ingest all EO data or all pointcloud data in a catalog, but even this wouldn't help. An individual record has no indication what 'type' it is, and an 'extension' isn't synonymous with 'type' anyway.

We can wait on this for now and discuss more later, I'll change the label.

**Comment by m-mohr** [comment]
Thanks for the response.
Your proposal sounds a little related to #264 as it both involves linking schemas.
It seems also a little related to #214 as you mentioned the type - but that only gives an indirect hint as you also noticed.
Yes, I also think we need more time to think about this. Personally, I am not sure yet what a good solution would look like.

**Comment by cholmes** [comment]
Yeah, this could be a good one to talk through - understand the use cases, etc.

I do think there are two types of extensions that we might treat differently. EO is a content extension - just different fields, and can be validated with a JSON Schema. Commons is more of a functionality or behavioral extension. 

For content extensions I would see an item articulating what content extensions it implements. And to me that would ideally link to a JSON Schema. Like it should refer to the extension name / id, as well as the schema that defines the extension. And it would link to all the schemas that define it. 

Matt, as you're putting implementations up feel free to just try out whatever idea makes sense to you and see how it goes in implementation. And then we can see how STAC Browser might make use of them to give some feedback, etc.

**Comment by m-mohr** [comment]
Regarding the recent discussion about Extension versioning: I think we shouldn't aim for separate version fields like eo:version and sci:version.
I think it is better solved in combination with this issue so that each specified extension could also hold the version number.

**Comment by m-mohr** [comment]
On STAC sprint 4 we talked a bit about it. Proposal was to basically just refer to JSON schema files, but allow shortcuts (the folder name?!) for core extensions. Example for a STAC Item (we could also put it in catalogs):

```json
{
  "stac_extensions": [
    "eo",
    "scientific",
    "cube",
    "checksum",
    "http://www.openeo.org/stac/0.5/processing-extension.json",
    "http://example.com/stac/1.0.1/landsat-extension.json"
  ]
}
```

As we don't really have versioning for extensions, we expect the core extensions to be the same version as the STAC spec. If not, it can be expressed by links to versioned JSON schemas of the core spec. For example: https://raw.githubusercontent.com/radiantearth/stac-spec/v0.6.2/extensions/eo/schema.json could be used in a STAC 0.7 Item (not recommended though). Opinions? @matthewhanson @jbants

**Comment by m-mohr** [comment]
Made a PR for discussions: #278.

**Comment by m-mohr** [comment]
Merged to dev, will be part of 0.8.0.
